specs/latest/2.0/index.html

```diff
@@ -1457,34 +1568,49 @@ <h2 class="no-toc">Table of contents</h2>
+                               GLsizei width, GLsizei height, GLenum format, GLenum type,
+                               ArrayBufferView srcData, GLuint srcOffset)
         <span class="gl-spec">(<a href="http://www.khronos.org/registry/gles/specs/3.0/es_spec_3.0.4.pdf#nameddest=section-3.8.5">OpenGL ES 3.0.4 &sect;3.8.5</a>, <a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexSubImage2D.xhtml">man page</a>)</span>
       </dt>
       <dd>
         <p>Only differences from <a href="../1.0/index.html#TEXSUBIMAGE2D">texSubImage2D</a> in WebGL 1 are described here. </p>
         <p>If a WebGLBuffer is bound to the <code>PIXEL_UNPACK_BUFFER</code> target, generates an <code>INVALID_OPERATION</code> error.</p>
         <p>The combination of <em>format</em>, <em>type</em>, and WebGLTexture's internal format must be listed in Table 1 or 2 from <span class="gl-spec"><a class="nonnormative" href="https://www.khronos.org/opengles/sdk/docs/man3/html/glTexImage2D.xhtml">man page</a></span>.</p>
-        <p>If <em>type</em> is <code>FLOAT_32_UNSIGNED_INT_24_8_REV</code>, generates an <code>INVALID_ENUM</code> error.</p>
-        <p>If <em>pixels</em> is non-null, the type of <em>pixels</em> must
-match the <em>type</em> according to the <a href="#TEXTURE_PIXELS_TYPE_TABLE">above table</a>; otherwise, generates an <code>INVALID_OPERATION</code> error.</p>
+        <p>If <em>srcData</em> is an ArrayBufferView, the type of <em>pixels</em> must
+           match the <em>type</em> according to the <a href="#TEXTURE_PIXELS_TYPE_TABLE">
+           above table</a>; otherwise, generates an <code>INVALID_OPERATION</code> error.</p>
         <p>See <a href="../1.0/index.html#PIXEL_STORAGE_PARAMETERS">Pixel Storage Parameters</a> for WebGL-specific pixel storage parameters that affect the behavior of this function.</p>
         <p>
             If <a href="#PIXEL_STORE_PARAM_CONSTRAINTS"> pixel store parameter constraints</a> are not met,
             generates an <code>INVALID_OPERATION</code> error.
         </p>
+        <p>Reading from <code>srcData</code> begins <code>srcOffset</code> elements into
+           <code>srcData</code>. (Elements are bytes for ArrayBuffer, int32s for Int32Array, etc.)
+        <p>If there's not enough data in <code>srcData</code> starting at <code>srcOffset</code>,
+           generate INVALID_OPERATION.
       </dd>
 
 
-      <dt><p class="idl-code"><a name="TEXSUBIMAGE2D_HTML">void texSubImage2D</a>(GLenum target, GLint level, GLint xoffset, GLint yoffset, GLenum format, GLenum type, TexImageSource? source) /* May throw DOMException */
+      <dt><p class="idl-code"><a name="TEXSUBIMAGE2D_HTML">
+        void texSubImage2D</a>(GLenum target, GLint level, GLint xoffset, GLint yoffset,
+                               GLsizei width, GLsizei height, GLenum format, GLenum type,
+                               (GLintptr or TexImageSource source) // May throw DOMException
```

Comment:
The union type for "source" should be removed, and just TexImageSource used. The overload below covers the case of uploading from pixel buffer objects.